bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/communication/SonnenJSONCommunication.java

```diff
@@ -38,21 +39,64 @@ public class SonnenJSONCommunication {
 
     private Gson gson;
     private @Nullable SonnenJsonDataDTO batteryData;
+    private SonnenJsonPowerMeterDataDTO @Nullable [] powerMeterData;
 
     public SonnenJSONCommunication() {
         gson = new Gson();
         config = new SonnenConfiguration();
     }
 
     /**
-     * Refreshes the battery connection.
+     * Refreshes the battery connection with the new API Call V2.
      *
      * @return an empty string if no error occurred, the error message otherwise.
      */
-    public String refreshBatteryConnection() {
+    public String refreshBatteryConnectionAPICALLV2(boolean powerMeter) {
         String result = "";
-        String urlStr = "http://" + config.hostIP + "/api/v1/status";
+        String urlStr = "http://" + config.hostIP + "/api/v2/status";
+        Properties httpHeader = new Properties();
+        httpHeader = createHeader(config.authToken);
+        try {
+            String response = HttpUtil.executeUrl("GET", urlStr, httpHeader, null, "application/json", 10000);
+            logger.debug("BatteryData = {}", response);
+            if (response == null) {
+                throw new IOException("HttpUtil.executeUrl returned null");
+            }
+            batteryData = gson.fromJson(response, SonnenJsonDataDTO.class);
+
+            if (powerMeter) {
+                response = HttpUtil.executeUrl("GET", "http://" + config.hostIP + "/api/v2/powermeter", httpHeader,
+                        null, "application/json", 10000);
+                logger.debug("PowerMeterData = {}", response);
+                if (response == null) {
+                    throw new IOException("HttpUtil.executeUrl returned null");
+                }
+
+                powerMeterData = gson.fromJson(response, SonnenJsonPowerMeterDataDTO[].class);
+            }
+        } catch (IOException | JsonSyntaxException e) {
+            logger.debug("Error processiong Get request {}:  {}", urlStr, e.getMessage());
+
+            if (e.getMessage() != null && e.getMessage().contains("WWW-Authenticate header")) {
```

Comment:
I'm wondering if the scenario being covered here is a 401 response without the WWW-Authenticate header, i.e. "java.util.concurrent.ExecutionException: org.eclipse.jetty.client.HttpResponseException: HTTP protocol violation: Authentication challenge without WWW-Authenticate header"?

In this case there might be a more direct way to approach this, since comparing the exception message is relying on very specific internal details of Jetty which might change at any time.

If this is indeed the scenario, you might find inspiration here:
https://github.com/openhab/openhab-addons/blob/6dfd6b01208f68592e9bc254bbdcbfd238d8bde7/bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/IndegoController.java#L184-L198

Reply:
I tried but cause is not an instance of HttpResponseException and never goes inside.

Reply:
`HttpUtil` wraps the exception in `IOException`:
https://github.com/openhab/openhab-core/blob/f92424c0b8c1b586d6c54a0345800049b8e48445/bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/HttpUtil.java#L254-L270

So it should be available, just further chained. However, relying on this for the check would also be vulnerable to changes. Without refactoring into using a dedicated `HttpClient` that could be configured to loosen up on the RFC 7235 violation, I don't see any really good solutions, so let's leave it as is.